services/credentials-service.ts

```diff
@@ -0,0 +1,62 @@
+import * as path from "path";
+import * as crypto from "crypto";
+
+export class CredentialsService implements ICredentialsService {
+	private static USERNAME_REGEX = /username\s+:\s(.*)/i;
+	private static PASSWORD_REGEX = /password\s+:\s(.*)/i;
+	private static ENCRYPTION_ALGORITHM = "aes-256-ctr";
+	private static ENCRYPTION_KEY = "6Gsz97KBp293Q0r0ei4pix98V4PIhm2D";
+	private pathToWindowsHelperScript: string;
+
+	constructor(private $childProcess: IChildProcess,
+		private $hostInfo: IHostInfo) {
+		this.pathToWindowsHelperScript = path.join(__dirname, "..", "vendor", "WinCredentialsHelper.ps1");
+	}
+
+	public async setCredentials(key: string, credentials: ICredentials): Promise<ICredentials> {
+		if (this.$hostInfo.isWindows) {
+			await this.$childProcess.spawnFromEvent("powershell.exe", [this.pathToWindowsHelperScript, "-AddCred", "-Target", key, "-User", credentials.username, "-Pass", this.encrypt(credentials.password)], "close");
+			return credentials;
+		} else {
+			throw new Error("Not implemented");
```

Comment:
I think the message here should be something like `Storing credentials is not supported on {osName} yet.`